config-proxy/src/main/java/com/yahoo/vespa/config/proxy/filedistribution/FileDownloader.java

```diff
@@ -0,0 +1,98 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.config.proxy.filedistribution;
+
+import com.google.common.collect.ImmutableSet;
+import com.yahoo.config.FileReference;
+import com.yahoo.config.subscription.ConfigSourceSet;
+import com.yahoo.log.LogLevel;
+import com.yahoo.vespa.defaults.Defaults;
+
+import java.io.File;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.*;
+import java.util.logging.Logger;
+
+/**
+ * Keeps track of files to download and download status
+ *
+ * @author hmusum
+ */
+public class FileDownloader {
+    private final static Logger log = Logger.getLogger(FileDownloader.class.getName());
+
+
+    private final String filesDirectory;
+    private final ConfigSourceSet configSourceSet;
+    private final Duration timeout;
+    private final Map<FileReference, Double> downloadStatus = new HashMap<>();
+    private final Set<FileReference> queuedForDownload = new LinkedHashSet<>();
```

Comment:
I assume locking some sort of locking is required here.

Reply:
Yes, just a placeholder, actually.